### ISSUE TITLE
Jobs Created not getting queued and submitted

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#8137]
https://its.cern.ch/jira/browse/CMSCOMPPR-1253

https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_SMP-PhaseIITDRSpring17DR-00040__v1_T_170808_173945_7603

the workflow is high priority already and there are 44 created jobs, only 41 were queued and submitted.

Those are the *next jobs* to be taken care of in the agent 0126 for quite some days now, so something is wrong, I would like to know what.
This concerns several upgrade requests that are just wait for a couple of jobs for many days now, so I raise it highest priority

**Comment by vlimant** [comment]
https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_MUO-PhaseIITDRSpring17DR-00011__v1_T_170806_191735_3600 is another one in that situation
https://its.cern.ch/jira/browse/CMSCOMPPR-1265
force completing because it cannot wait for debugging. -unsatisfactory for permanent operation-

**Comment by amaltaro** [comment]
When a job is created, it's automatically queued in the agent (it's created in the agent queue).

pdmvserv_task_SMP-PhaseIITDRSpring17DR-00040__v1_T_170808_173945_7603
has 0 jobs queued or active. I see only failures and success.

And since the second one was force-completed, I think there is nothing left for this ticket.

As I said in the JIRA ticket, the agent was behaving as expected. Sites were providing almost 0 slots to this agent and there were no free pending thresholds. And when there was, logcollect, cleanup and merge jobs could be running before these 190k prio processing jobs. Perhaps we should remove the multiplier factor for those jobs and simply bump their prio by 50k or so?

Anyways, I suggest we follow that issue on https://github.com/dmwm/WMCore/issues/8129

**Comment by vlimant** [comment]
I can read

SMP-PhaseIITDRSpring17MiniAOD-00040_0MergeMINIAODSIMoutput | 44 | 0 | 0 | 0 | 41 | 0 | 0 | 91.9% | 91.9%
-- | -- | -- | -- | -- | -- | -- | -- | -- | --

which means there are potentially 3 jobs (44-41) that are "created" and nowhere to be found.
if there is nothing left for the workflow to be done, why is it not getting into "completed".

**Comment by vlimant** [comment]
same for 

SMP-PhaseIITDRSpring17DR-00040_1RECOSIMoutputMergeLogCollect | 34 | 0 | 0 | 0 | 0 | 31 | 0 | 0.0% | 0.0%
-- | -- | -- | -- | -- | -- | -- | -- | -- | --



LogCollectForSMP-PhaseIITDRSpring17MiniAOD-00040_0 | 29 | 0 | 0 | 0 | 0 | 26 | 0 | 0.0% | 0.0%
-- | -- | -- | -- | -- | -- | -- | -- | -- | --


which I don't care about (logcollect) but shows that there is something wrong (in what one can view)

**Comment by vlimant** [comment]
agreed on #8129 which indeed was the case, and got unstuck with killing that big workflow at 63k priority. but then after that it should have gone all the way ; and it's not the case.

**Comment by amaltaro** [comment]
It turns out JobAccountant had been terminated around 2 days ago (2017-09-05 06:24:08,336). I've restarted it and it's slowly going through 170k jobs now.

I see _7603 has been moved to completed already. So I'm closing this issue.

Just a note, the next WMAgent release should have a better component monitoring, we've done some improvements to the component heartbeat and in case there is none for 30min, then it will update WMStats with a component Error. I believe a very busy system can still yield false alarms, but we'll learn with that.
